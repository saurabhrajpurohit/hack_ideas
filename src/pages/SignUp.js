import React from "react";

import { Button, Link, TextField } from "@mui/material";
import { UserService } from "../service/Service";

const SignUp = () => {
    const [employeeID, setEmployeeID] = React.useState("");
    const [name, setName] = React.useState("");

    const submit = (event) => {
        event.preventDefault();
        UserService.getByField("EmployeeID", employeeID).then(data => console.log(data));
    }

    return (
        <form onSubmit={submit}>
            <TextField id="employeeID" value={employeeID} variant="outlined" label="Employee ID" required onChange={(event) => setEmployeeID(event.target.value)} />
            <TextField id="name" value={name} variant="outlined" label="Employee ID" required onChange={(event) => setName(event.target.value)} />
            <Button id="submit" variant="contained" color="primary" type="submit">Sign Up</Button>
            <Link href="/home">Go Back</Link>
        </form>
    )
}

export default SignUp;