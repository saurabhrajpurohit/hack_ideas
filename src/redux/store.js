import { createStore } from "redux";
import { SET_USER } from "../utils/constants";

const userState = {};

const userReducer = (state = userState, action) => {
    switch (action.type) {
        case SET_USER:
            return {
                ...state,
                user: action.value
            }
        default:
            return state;
    }
};

export default createStore(userReducer);

