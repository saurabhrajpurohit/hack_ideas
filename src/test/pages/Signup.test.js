import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import SignUp from '../../pages/SignUp';
import store from '../../redux/store';

test('ui to be rendered correctly', () => {
    render(
        <Provider store={store}>
            <BrowserRouter>
                <SignUp />
            </BrowserRouter>
        </Provider>
    );
    const element = screen.getByTestId('signup-form');
    expect(element).toMatchSnapshot();
});

test('ui to be rendered with errors', () => {
    render(
        <Provider store={store}>
            <BrowserRouter>
                <SignUp />
            </BrowserRouter>
        </Provider>
    );
    const button = screen.getByTestId('signup-submit');
    fireEvent.click(button);
    const element = screen.getByTestId('signup-form');
    expect(element).toMatchSnapshot();
});
