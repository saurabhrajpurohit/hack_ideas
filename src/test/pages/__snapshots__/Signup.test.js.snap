// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui to be rendered correctly 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="signup-form"
>
  <header
    class="css-sg2wnk"
  >
    <a
      href="/"
    >
      <div
        class="MuiBox-root css-1yb5pbx"
      >
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom css-1hleqvg-MuiTypography-root"
        >
          Hack 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmojiObjectsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-.46 0-.93.04-1.4.14-2.76.53-4.96 2.76-5.48 5.52-.48 2.61.48 5.01 2.22 6.56.43.38.66.91.66 1.47V19c0 1.1.9 2 2 2h.28c.35.6.98 1 1.72 1s1.38-.4 1.72-1H14c1.1 0 2-.9 2-2v-2.31c0-.55.22-1.09.64-1.46C18.09 13.95 19 12.08 19 10c0-3.87-3.13-7-7-7zm2 16h-4v-1h4v1zm0-2h-4v-1h4v1zm-1.5-5.59V14h-1v-2.59L9.67 9.59l.71-.71L12 10.5l1.62-1.62.71.71-1.83 1.82z"
            />
          </svg>
           Idea's
        </span>
      </div>
    </a>
    <p
      class="MuiTypography-root MuiTypography-body2 css-p34ji4-MuiTypography-root"
    >
      Already have an account?  
      <a
        class="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineNone css-1ser5cc-MuiTypography-root-MuiLink-root"
        href="/login"
      >
        Login
      </a>
    </p>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="css-26q4qm"
    >
      <div
        class="css-lr1r4n-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
        >
          Get Started
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Enter your details below.
        </p>
      </div>
      <form>
        <div
          class="css-1lwbda4-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="employeeID"
              id="employeeID-label"
            >
              Employee ID
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="employeeID"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Employee ID
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="name"
              id="name-label"
            >
              Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="signup-submit"
            id="submit"
            tabindex="0"
            type="submit"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`ui to be rendered with errors 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="signup-form"
>
  <header
    class="css-sg2wnk"
  >
    <a
      href="/"
    >
      <div
        class="MuiBox-root css-1yb5pbx"
      >
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom css-1hleqvg-MuiTypography-root"
        >
          Hack 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmojiObjectsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-.46 0-.93.04-1.4.14-2.76.53-4.96 2.76-5.48 5.52-.48 2.61.48 5.01 2.22 6.56.43.38.66.91.66 1.47V19c0 1.1.9 2 2 2h.28c.35.6.98 1 1.72 1s1.38-.4 1.72-1H14c1.1 0 2-.9 2-2v-2.31c0-.55.22-1.09.64-1.46C18.09 13.95 19 12.08 19 10c0-3.87-3.13-7-7-7zm2 16h-4v-1h4v1zm0-2h-4v-1h4v1zm-1.5-5.59V14h-1v-2.59L9.67 9.59l.71-.71L12 10.5l1.62-1.62.71.71-1.83 1.82z"
            />
          </svg>
           Idea's
        </span>
      </div>
    </a>
    <p
      class="MuiTypography-root MuiTypography-body2 css-p34ji4-MuiTypography-root"
    >
      Already have an account?  
      <a
        class="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineNone css-1ser5cc-MuiTypography-root-MuiLink-root"
        href="/login"
      >
        Login
      </a>
    </p>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="css-26q4qm"
    >
      <div
        class="css-lr1r4n-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
        >
          Get Started
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Enter your details below.
        </p>
      </div>
      <form>
        <div
          class="css-1lwbda4-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="employeeID"
              id="employeeID-label"
            >
              Employee ID
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby="employeeID-helper-text"
                aria-invalid="true"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="employeeID"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Employee ID
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
              id="employeeID-helper-text"
            >
              Employee ID is required
            </p>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="name"
              id="name-label"
            >
              Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-describedby="name-helper-text"
                aria-invalid="true"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="name"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Name
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
              id="name-helper-text"
            >
              Name is required
            </p>
          </div>
          <div
            class="MuiBox-root css-8atqhb"
          >
            <span
              class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
              role="progressbar"
            >
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
              />
              <span
                class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
              />
            </span>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="signup-submit"
            id="submit"
            tabindex="0"
            type="submit"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
